ouroboros-consensus-test/src/Test/LedgerTables.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module Test.LedgerTables (
+    prop_stowable_laws
+  , prop_tablestuff_laws
+  ) where
+
+import           Data.Function (on)
+
+import           Ouroboros.Consensus.Ledger.Basics
+import           Ouroboros.Consensus.Ledger.Tables
+
+import           Test.QuickCheck
+
+compareTables ::
+     ( IsMapKind mk
+     , TableStuff (LedgerState blk)
+     , Eq (LedgerTables (LedgerState blk) mk)
+     , Show (LedgerTables (LedgerState blk) mk)
+     )
+  => LedgerState blk mk
+  -> LedgerState blk mk
+  -> Property
+compareTables = (===) `on` projectLedgerTables
+
+prop_stowable_laws ::
+     ( TableStuff (LedgerState blk)
+     , StowableLedgerTables (LedgerState blk)
+     , Eq (LedgerTables (LedgerState blk) EmptyMK)
+     , Show (LedgerTables (LedgerState blk) EmptyMK)
+     , Show (LedgerTables (LedgerState blk) ValuesMK)
+     )
+  => LedgerState blk EmptyMK
+  -> Property
+prop_stowable_laws = \ls ->
+  compareTables (stowLedgerTables (unstowLedgerTables ls)) ls .&&.
+  let ls' = unstowLedgerTables ls
+  in compareTables (unstowLedgerTables (stowLedgerTables ls')) ls'
+
+prop_tablestuff_laws ::
+     ( TableStuff (LedgerState blk)
+     , Eq (LedgerTables (LedgerState blk) EmptyMK)
+     , Show (LedgerTables (LedgerState blk) EmptyMK)
+     , Show (LedgerTables (LedgerState blk) ValuesMK)
+     )
+  => LedgerState blk EmptyMK
+  -> LedgerTables (LedgerState blk) ValuesMK
+  -> Property
+prop_tablestuff_laws = \ls tbs ->
+  compareTables (ls `withLedgerTables` (projectLedgerTables ls)) ls .&&.
```

Comment:
Why not compare the ledger states instead of the ledger tables? This would give a stronger property

Reply:
I added a comment explaining why. Basically the ledger state can be arbitrarily complex/big and comparing the whole ledger state takes a non-trivial amount of time more. But it can be done

Reply:
But besides the runtime cost, isn't comparing /just the ledger tables/ the right thing to do in this case?

Reply:
>But besides the runtime cost, isn't comparing /just the ledger tables/ the right thing to do in this case?

Since we're testing `withLedgerTables . projectLedgerTabels == id`, I can agree that it is the right thing to do. Still, maybe another law could be added which states that `withLedgerTables` should *only* change the ledger tables and not any other part of the ledger state

Reply:
I will keep this as is. We can add the full check later.